### PR TITLE
Bridge Integration

### PR DESCRIPTION
**Pull request by mystbrent**
## Description of Changes

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-183)
[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-181)

1. Token Approvals Integration
2. Deposit Bridge Integration
3. Burn Wrapped Tokens Integration
4. Switching Bridge Origin/Destination feature
5. Choosing Token to Bridge feature
6. Button state changes react to current balance/allowance

### How to test:
1. Update you env file from [1Password](https://start.1password.com/open/i?a=2VIA62LB7VAW3NACLOHTFW4ERI&h=halodao.1password.com&i=6bzbszvhwrbm7lztrvp5sfhfy4&v=gllrg2so74fsgd43baj5q5voui) - (If you're testing locally)
2. Secure MKT (Mock Token) from Brent. 
3. Secure Matic tokens for your gas -> https://matic.supply/
3. Go to /Bridge.
4. Make sure you are in Polygon Mainnet.
5. Default network will be selected (in this case Polygon -> BSC).
6. Enter amount of MKT you want to bridge.
7. Approve button should appear.
8. Once approve tx is confirmed, "Next" button will appear, click "Next" to bridge the token.
9. Confirm the tx in Metamask.
10. Confirm transaction modal should appear.
11. When transaction is confirmed, your MKT balance is polygon should be updated.
12. To check your WMKT balance, select BSC as the source network.
13. WMKT should be selected from the currency input.
14. Your WMKT balance should be updated only after the transaction is considered final. (After 192 blocks for polygon - you can check polygonscan)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [ ] All files have appropriate file headers and documentation
* [ ] I have written new tests for your core changes, as applicable
* [ ] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

File: src/components/Tailwind/Modals/TokenSelectModal.tsx
Comment (0xAplki) [inline src/components/Tailwind/Modals/TokenSelectModal.tsx]: eslint warning here

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: eslint warning here

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: here also

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: here also, maybe you can check on using `useCallback` also for this

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: same here

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: here also

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: eslint here

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: eslint here

File: src/pages/Tailwind/Bridge/index.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/index.tsx]: eslint here

**Review by 0xAplki — CHANGES_REQUESTED** [review]
Just eslint stuff and some suggestions. otherwise, looking good!

**Comment by mystbrent** [comment]
Fixed earlier call's comments @tracyarciaga :D

File: src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx
Comment (0xAplki) [inline src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx:14]: Last eslint here @mystbrent . approving this now but pls fix before merge :)

**Review by 0xAplki — APPROVED** [review]
Left one comment. approving this but fix it before we merge. thanks @mystbrent !

**Review by tracyarciaga — COMMENTED** [review]
Few minor UI flow comments @mystbrent 

1. On the SuccessContent of the BridgeTransactionModal, please make these dynamic depending on the selected network (unless @schystz already did something similar that we can resuse?):
https://github.com/HaloDAO/halodao-interface/blob/feature/hdev-183/src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx#L177

https://github.com/HaloDAO/halodao-interface/blob/feature/hdev-183/src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx#L181 

2. After transaction confirmation, reset the InputValue.

File: src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx
Comment (tracyarciaga) [inline src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx]: @mystbrent Is this ok to be 0?

File: src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx
Comment (tracyarciaga) [inline src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx]: @mystbrent Is this ok to be 0? Is this suppose to be our fee? I think we should include our fee in this modal.

File: src/constants/bridgeAbis/PrimaryBridge.json
Comment (schystz) [inline src/constants/bridgeAbis/PrimaryBridge.json]: Can we move this file under `haloAbis`? We wanna keep all ABIs we are using inside the subfolder to separate it from sushi/uni contracts

File: src/constants/bridgeAbis/SecondaryBridge.json
Comment (schystz) [inline src/constants/bridgeAbis/SecondaryBridge.json]: Can we move this file under `haloAbis`? We wanna keep all ABIs we are using inside the subfolder to separate it from sushi/uni contracts

File: src/constants/index.ts
Comment (schystz) [inline src/constants/index.ts]: Seems to be unused

File: src/constants/index.ts
Comment (schystz) [inline src/constants/index.ts]: Seems to be unused

**Review by schystz — COMMENTED** [review]
Initial review. Please find below minor comments.

Also noticed there are a lot of ESLint warnings on the Files change tab. Let's fix those as well.

The Bridge component is also pretty big, maybe we can make it more modular so it is also easier to review and test (eventually). Though I also think it will be easier to demo the functionality over a call, and just to briefly summarize the building blocks for bridge in the code level. Let me me know your thoughts @mystbrent

File: src/components/Tailwind/Modals/TokenSelectModal.tsx
Comment (schystz) [inline src/components/Tailwind/Modals/TokenSelectModal.tsx:58]: @mystbrent search functionality (lines 52-56) is not implemented yet

Not sure if we wanna show that everytime though, maybe if its just 5 tokens or less we don't have to show it since user can easily see all tokens without scrolling.

Comment (mystbrent) [reply src/components/Tailwind/Modals/TokenSelectModal.tsx:58]: Yeah makes sense. Tbh fyi also right now though the list of supported tokens are set in constant

**Comment by mystbrent** [comment]
> Initial review. Please find below minor comments.
> 
> Also noticed there are a lot of ESLint warnings on the Files change tab. Let's fix those as well.
> 
> The Bridge component is also pretty big, maybe we can make it more modular so it is also easier to review and test (eventually). Though I also think it will be easier to demo the functionality over a call, and just to briefly summarize the building blocks for bridge in the code level. Let me me know your thoughts @mystbrent

Thank you for checking @schystz 

Yeah happy to demo this let me know when you might be available so that we can arrange.

I have also something to ask/discuss with the code

File: src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx
Comment (mystbrent) [reply src/pages/Tailwind/Bridge/modals/BridgeTransactionModal.tsx]: Thanks for the find.

I am actually confused with what is the difference between shuttle fee and tax perhaps you can explain more about it?

**Comment by tracyarciaga** [comment]
Additional minor issues I found: @mystbrent 
1. ✅ DONE - Remove 0 as the default value in currencyInput when bridge page is loaded, this could be a minor UI flow issue. Check video below:  

https://user-images.githubusercontent.com/6169194/130786374-355070ac-4a08-43d0-8b9c-0ec45ca77eb4.mov

2. Seems that approve spent is getting reset after a successful tx. I inititally approved 10MKT, and proceeded to bridge it and tx was successful. But when I tried to bridge again for <=10MKT , still it's asking me to approve. (it shouldn't)
3. ✅ DONE - Gas fee is not formatted
image

4. ✅ DONE -Should add text to inform user about tx fiinality (Maybe in extra info)

**Pull request closed without merging**
